demos/bar-heatmap.html

```diff
@@ -0,0 +1,330 @@
+				}
+
+				function setCursor(u) {
+					const {left, top, idx} = u.cursor;
+
+					let tt = u.seriestt[1]
+					let s = u.series[1];
+
+					let xVal = u.data[0][idx];
+					let yVal = u.data[1][idx];
+					if (yVal == null) return;
+					let zVal = u.data[2][idx];
+
+					tt.textContent = "-(" + yVal + " hits:" + zVal + ")-";
+
+					tt.style.left = Math.round(u.valToPos(xVal, 'x')) + "px";
+					tt.style.top = Math.round(u.valToPos(yVal, s.scale)) + "px";
+					tt.style.background = "black";
+					tt.style.color = "white";
+
+					u.cursortt.style.left = left + "px";
+					u.cursortt.style.top = top + "px";
+
+					// OLD - basic display
+					u.cursortt.textContent = "( " + Math.round(top, "y") + ")";
+					// scan the series data for a hit
+					let liveYValue = Math.round(u.posToVal(top, "y"));
+
+					let foundIndex = yVal.indexOf(liveYValue);
+					if (foundIndex >= 0){
+						u.cursortt.textContent = "- Latency:" +liveYValue + " Hits: " + zVal[foundIndex] + " -";
+					}
+				}
+
+				return {
+					hooks: {
+						init,
+						setCursor,
+						setScale: [
+							(u, key) => {
+								console.log('setScale', key);
+							}
+						],
+						setSeries: [
+							(u, idx) => {
+								console.log('setSeries', idx);
+							}
+						],
+					},
+				};
+			}
+
+
+
+			function ladderRenderPlugin({ gap = 2, shadowColor = "#000000", bodyMaxWidth = 50 } = {}) {
+
+				function renderLadderY(ladderY, index) {
+						let ladderAsY = this.u.valToPos(ladderY,   "y", true);
+
+						let ladderValue = this.u.data[2][this.i][index]
+						let ladderPercent = ladderValue / this.maxHeatValue;
+
+						let colorHeat = parseInt(255 * ladderPercent);
+					   this.u.ctx.fillStyle = "rgba(50,50," + colorHeat + ",0.5)"
+
+						this.u.ctx.fillRect(
+							Math.round(this.bodyX),
+							Math.round(ladderAsY),
+							Math.round(this.bodyWidth),
+							Math.round(5),
+						);
+				}
+
+				function drawSeries(u) {
+					u.ctx.save();
+
+					let [iMin, iMax] = u.series[0].idxs;
+					let columnWidth  = u.bbox.width / (iMax - iMin);
+					let bodyWidth    = Math.min(bodyMaxWidth, columnWidth - gap);
+
+					for (let i = iMin; i <= iMax; i++) {
+						let xVal         = u.scales.x.distr == 2 ? i : u.data[0][i];
+						let timeAsX      = u.valToPos(xVal,  "x", true);
+						let bodyX        = timeAsX - (bodyWidth / 2);
+						let ladderY         = u.data[1][i];
+
+						ladderY.forEach(renderLadderY.bind({ i:i, u:u, xVal:xVal, bodyX:bodyX, bodyWidth:bodyWidth, maxHeatValue: u.data[3][1]}));
+					}
+					u.ctx.restore();
+				}
+
+				return {
+					hooks: {
+						draw: drawSeries,
+					}
+				};
+			}
+
+			const data = [
+				// dates
+				[1546300800,1546387200,1546473600,1546560000,1546819200,1546905600,1546992000,1547078400,1547164800,1547424000,1547510400,1547596800],
+
+				//  latency ladder
+				[
+					[80,  50,  43],
+					[10,  12,  14,  18,  30,  50],
+					[70,  90,  13,  10,  12,  13],
+					[170, 190, 113, 110, 112, 113],
```

Comment:
@bluemonk3y i'm trying to plug your data into my prototype and found this bit confusing, since i'm assuming that the latency values are unique and their counts would appear summed together. is there a specific reason for having 113 repeat here? the original description [1] appears to outline what i ended up with in my prototype so far, which would not have repeated latencies (or buckets).

(also, latencies in my dataset are sorted ascending for easier hover testing, so that'll also be a requirement of the input format).

[1] https://github.com/liquidlabsio/fluidity/issues/65#issuecomment-625448053